### PR TITLE
Csv to single text conversation

### PR DESCRIPTION
**Pull request by nipunaprs**
initial script for taking large csvs with call logs and converting to single utterance.

### CODE REVIEW

File: csv_to_json_large_convos_to_single_utterance.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
+"""
+python csv_to_json_combine_large_convos.py
```
Comment (faya98) [inline csv_to_json_large_convos_to_single_utterance.py:2]: Doesn't match the file name

File: csv_to_json_large_convos_to_single_utterance.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+"""
+# *********************************************************************************************************************
+import click
```
Comment (faya98) [inline csv_to_json_large_convos_to_single_utterance.py:6]: Split the libraries into standard, 3rd party and custom imports. For example, see https://github.com/zia-ai/academy/blob/master/ada_to_csv.py

File: csv_very_large_file.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+"""
+python csv_to_json_very_large_file.py
```
Comment (faya98) [inline csv_very_large_file.py:2]: File name doesn't match

File: csv_to_json_large_convos_to_single_utterance.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        file_name_split = file_name
+
+    json_file_name = file_name_split + "-hf-export.json"
+
+    convo_dict = combine_conversations(filename,utterance_col,convo_id_col,created_at_col,unix_date,role_col,role_mapper)
+    export_to_hf_json_limit(convo_dict,json_file_name,limit=limit)
+
+
+
+def combine_conversations(filename: str,utterance_col,convo_id_col,created_at_col,unix_date,role_col,role_mapper):
```
Comment (faya98) [inline csv_to_json_large_convos_to_single_utterance.py:50]: provide the data type of every function variables and also the data type of the values returned. If no values are returned by the function, then provide None

File: csv_to_json_large_convos_to_single_utterance.py [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+        with open(err_file_name, 'w', newline='') as err_csv_file:
+            writer = csv.writer(err_csv_file)
+            writer.writerows(error_log)
+
+
+    return convo_dict
+
+
+def export_to_hf_json(conversation_dict, json_file):
```
Comment (faya98) [inline csv_to_json_large_convos_to_single_utterance.py:183]: Add docstring to the function explaining what the function does.

File: csv_to_json_large_convos_to_single_utterance.py [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+    data = {
+        "$schema": "https://docs.humanfirst.ai/hf-json-schema.json",
+        "examples": examples
+    }
+
+    with open(json_file, 'w', encoding='utf-8') as jsonfile:
+        json.dump(data, jsonfile, indent=2)
+
+def export_to_hf_json_limit(conversation_dict, json_file, limit):
```
Comment (faya98) [inline csv_to_json_large_convos_to_single_utterance.py:204]: Add docstring to the function explaining what the function does.

File: csv_very_large_file.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+              help='If conversations which is the id otherwise utterances and defaults to hash of utterance_col')
+@click.option('-t', '--created_at_col', type=str, required=False, default='',
+              help='If there is a created date for utterance otherwise defaults to now')
+@click.option('-x', '--unix_date', is_flag=True, type=bool, required=False, default=False,
+              help='If created_at column is in unix epoch format')
+@click.option('-r', '--role_col', type=str, required=False, default='',
+              help='Which column the role in ')
+@click.option('-p', '--role_mapper', type=str, required=False, default='',
+              help='If role column then role mapper in format "source_client:client,source_expert:expert,*:expert"')
+def main(filename: str):
```
Comment (faya98) [inline csv_very_large_file.py:25]: Provide the data type of the values returned. If no values are returned by the function, then provide None

Add docstring to the function explaining what the function does.

File: csv_very_large_file.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    file_name = filename.split("/")[-1]
+    file_name_split = file_name.split(".")[0]
+    json_file_name = file_name_split + "-hf-export.json"
+    convo_dict = combine_conversations(file_name)
+    export_to_hf_json_limit(convo_dict,json_file_name,6000)
+
+
+def combine_conversations(filename: str):
```
Comment (faya98) [inline csv_very_large_file.py:33]: Provide the data type of the values returned. If no values are returned by the function, then provide None

Add docstring to the function explaining what the function does.

File: csv_very_large_file.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+    data = {
+        "$schema": "https://docs.humanfirst.ai/hf-json-schema.json",
+        "examples": examples
+    }
+
+    with open(json_file, 'w', encoding='utf-8') as jsonfile:
+        json.dump(data, jsonfile, indent=2)
+
+def export_to_hf_json_limit(conversation_dict, json_file, limit):
```
Comment (faya98) [inline csv_very_large_file.py:105]: Provide the data type of every function variables and also the data type of the values returned. If no values are returned by the function, then provide None

Add docstring to the function explaining what the function does.

**Review by nipunaprs — COMMENTED** [review]
Submitting commit to review

**Review by faya98 — APPROVED** [review]
Nice work!

**Pull request merged**